### DISCUSSION TITLE
Dynamic Paths from data troubles

### DISCUSSION THREAD

**Discussion by Kurohyou** [vuejs/vitepress#2918] [Q&A] [answered]
I'm trying to build documentation for a library of functions I built and maintain. The end goal is to create an API documentation site where the various exposed methods, properties, and functions can be documented with their parameters, return values, descriptions, and example code; something similar to say the [chrome.tabs API method reference](https://developer.chrome.com/docs/extensions/reference/tabs/#method). But I'm having difficulty getting vitepress to generate dynamic paths from my data

The error I get is:
```
failed to load config from @\reference\pug\[name].paths.js
Invalid paths file export in @/reference/pug/[name].paths.js. Expects default export of an object with a "paths" property.
```
But, my paths file does have a paths property in the export:

```js
import { pug } from '@/data'

export default {
  paths() {
    return pug.map(
      obj => (
        {
          params:{
            name:obj.meta.name,
            ...obj
          }
        }
      )
    )
  }
}
```

Any ideas on what I'm doing wrong here? Or am I trying to use dynamic paths for something they aren't meant for?

**Comment by brc-dd** [comment, marked as answer]
Aliases aren't supported inside data loader, path and config files. Try using relative imports instead.

**Reply by Kurohyou** [reply]
Ok, that does get me past that block. Is that documented somewhere?

Thanks for the help!

Now to figure out why it's trying to generate a file with line ending designation at the end of the file name instead of `.md`.
